system/handlers/formcontrols/ReadOnly.cfc

```diff
@@ -0,0 +1,22 @@
+component {
+
+	property name="presideObjectService"   inject="presideObjectService";
+	property name="contentRendererService" inject="contentRendererService";
+
+	public string function index( event, rc, prc, args={} ) {
+		var properties = presideObjectService.getObjectProperties( args.sourceObject );
+		var prop       = properties[ args.name ] ?: {};
+		
+		args.renderer  = args.renderer ?: contentRendererService.getRendererForField( prop );
+
+		if ( !contentRendererService.rendererExists( args.renderer ) ) {
+			args.renderer = type;
```

Comment:
`type` is not defined I think? I think this should be `""`? Can then also ditch the `renderer == "none"` check below.